synapse/state/v1.py

```diff
@@ -329,12 +329,10 @@ def _resolve_auth_events(
         auth_events[(prev_event.type, prev_event.state_key)] = prev_event
         try:
             # The signatures have already been checked at this point
-            event_auth.check(
+            event_auth.check_auth_rules_for_event(
```

Comment:
this means we skip the check that the event has a `room_id` attribute, but there's no way we could have got this far with an event without one. Similarly in a bunch of places below.